pkg/types/var.go

```diff
@@ -70,26 +70,45 @@ func (v *Var) defaulting() {
 	}
 }
 
-// VarSet is a slice of Vars where no var.Name is repeated.
+// VarSet is a set of Vars where no var.Name is repeated.
 type VarSet struct {
-	set []Var
+	set map[string]Var
+}
+
+// NewVarSet returns an initialized VarSet
+func NewVarSet() VarSet {
+	return VarSet{set: map[string]Var{}}
 }
 
 // AsSlice returns the vars as a slice.
 func (vs *VarSet) AsSlice() []Var {
 	s := make([]Var, len(vs.set))
-	copy(s, vs.set)
+	i := 0
+	for _, v := range vs.set {
+		s[i] = v
+		i++
+	}
+	sort.Sort(ByName(s))
```

Comment:
Please toss in a String() method that calls this method so we get deterministic ordering when debugging.